lib/model/message_list.dart

```diff
@@ -86,6 +87,30 @@ class MessageListView extends ChangeNotifier {
     notifyListeners();
   }
 
+  Message applyChangesToMessage(UpdateMessageEvent event, Message oldMessage) {
+    Map<String, dynamic> oldMessageJson = oldMessage.toJson();
+
+    //TODO: What other fields should be edited by the update event?
+    oldMessageJson["content"] = event.renderedContent;
+    oldMessageJson["lastEditTimestamp"] = event.editTimestamp;
+
+    return Message.fromJson(oldMessageJson);
+  }
+
+  void maybeUpdateMessage(UpdateMessageEvent event) {
+    int oldIdx = messages.indexWhere((m) => m.id == event.messageId);
+    if (oldIdx == -1)  {
+      return;
+    }
+
+    Message oldMessage = messages[oldIdx];
+    Message newMessage = applyChangesToMessage(event, oldMessage);
+
+    messages.replaceRange(oldIdx, oldIdx+1, [newMessage]);
+    contents.replaceRange(oldIdx, oldIdx+1, [parseContent(newMessage.content)]);
```

Comment:
These can be simpler than `replaceRange`. See the language tour:
https://dart.dev/language/collections#lists

Reply:
👍 Will fix